components/org.wso2.carbon.identity.api.user.push.device.handler.v1/src/main/java/org/wso2/carbon/identity/api/user/push/device/handler/v1/core/PushDeviceHandlerService.java

```diff
@@ -0,0 +1,280 @@
+        deviceDTO.setLastUsedTime(device.getLastUsedTime());
+        return deviceDTO;
+    }
+
+    /**
+     * Get a list of all the devices for the authenticated in user.
+     *
+     * @return List of registered devices of the user
+     */
+    public List<DeviceDTO> listDevices() {
+
+        List<Device> devices;
+        User user = getAuthenticatedUser();
+
+        try {
+            String userId = getUserIdFromUsername(user.getUserName());
+            devices = PushDeviceHandlerServiceHolder.getPushDeviceHandlerService().listDevices(userId);
+        } catch (PushDeviceHandlerServerException e) {
+            throw PushDeviceApiUtils.handleException(e,
+                    PushDeviceApiConstants.ErrorMessages.ERROR_CODE_LIST_DEVICE_SERVER_ERROR);
+        } catch (UserStoreException e) {
+            throw PushDeviceApiUtils.handleException(e,
+                    PushDeviceApiConstants.ErrorMessages.ERROR_CODE_USER_STORE_ERROR);
+        }
+        List<DeviceDTO> deviceDTOArrayList = new ArrayList<>();
+        DeviceDTO deviceDTO;
+        if (devices != null) {
+            for (Device device : devices) {
+                deviceDTO = new DeviceDTO();
+                deviceDTO.setDeviceId(device.getDeviceId());
+                deviceDTO.setName(device.getDeviceName());
+                deviceDTO.setModel(device.getDeviceModel());
+                deviceDTO.setRegistrationTime(device.getRegistrationTime());
+                deviceDTO.setLastUsedTime(device.getLastUsedTime());
+                deviceDTOArrayList.add(deviceDTO);
+            }
+        }
+        return deviceDTOArrayList;
+    }
+
+    /**
+     * Get discovery data for registering a new device.
+     *
+     * @return Discovery data
+     */
+    public DiscoveryDataDTO getDiscoveryData() {
+
+        RegistrationDiscoveryData discoveryData;
+        try {
+            discoveryData = PushDeviceHandlerServiceHolder.getPushDeviceHandlerService().getRegistrationDiscoveryData();
+        } catch (PushDeviceHandlerServerException e) {
+            throw PushDeviceApiUtils.handleException(e,
+                    PushDeviceApiConstants.ErrorMessages.ERROR_CODE_REGISTER_DEVICE_SERVER_ERROR);
+        }
+        DiscoveryDataDTO discoveryDataDTO = new DiscoveryDataDTO();
+        discoveryDataDTO.setDid(discoveryData.getDeviceId());
+        discoveryDataDTO.setUn(discoveryData.getUsername());
+        discoveryDataDTO.setTd(discoveryData.getTenantDomain());
+        discoveryDataDTO.setFn(discoveryData.getFirstName());
+        discoveryDataDTO.setLn(discoveryData.getLastName());
+        discoveryDataDTO.setChg(discoveryData.getChallenge());
+        discoveryDataDTO.setHst(discoveryData.getHost());
+        discoveryDataDTO.setBp(discoveryData.getBasePath());
+        discoveryDataDTO.setRe(discoveryData.getRegistrationEndpoint());
+        discoveryDataDTO.setRde(discoveryData.getRemoveDeviceEndpoint());
```

Comment:
Add comments explaining what the letters denote

Reply:
Not adding comments as the meanings can be seen in the strings passed to the setter